sdk/iot/Azure.Iot.Hub.Service/tests/ConfigurationsClientTests.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            const int configurationsCount = 5;
+            TwinConfiguration[] twinConfigurations = new TwinConfiguration[configurationsCount];
+            TwinConfiguration[] createdConfigurations = new TwinConfiguration[configurationsCount];
+            IReadOnlyList<TwinConfiguration> listConfigurations;
+            IotHubServiceClient client = GetClient();
+            try
+            {
+                for (int i = 0; i < configurationsCount; i++)
+                {
+                    twinConfigurations[i] = CreateTestConfig($"testconfigurations{i}{GetRandom()}");
+
+                    // Create Configurations
+                    Response<TwinConfiguration> createResponse =
+                        await client.Configurations.CreateOrUpdateConfigurationAsync(twinConfigurations[i]).ConfigureAwait(false);
+                    createdConfigurations[i] = createResponse.Value;
+                }
+
+                // List the configurations for the client
+                listConfigurations = (await client.Configurations.GetConfigurationsAsync().ConfigureAwait(false)).Value;
+
+                IEnumerable<string> twinConfigurationsIds = listConfigurations.ToList().Select(configuration => configuration.Id);
+
+                // Compare the response ids with created configurations
+                for (int i = 0; i < configurationsCount; i++)
+                {
+                    Assert.IsTrue(twinConfigurationsIds.Contains(twinConfigurations[i].Id));
+                }
+            }
+            finally
+            {
+                for (int i = 0; i < configurationsCount; i++)
+                    await Cleanup(client, twinConfigurations[i]);
+            }
+        }
+
+        private async Task Cleanup(IotHubServiceClient client, TwinConfiguration config)
+        {
+            // cleanup
+            try
+            {
+                if (config != null)
+                {
+                    await client.Configurations.DeleteConfigurationAsync(config, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail($"Test clean up failed: {ex.Message}");
+            }
+        }
+
+        private TwinConfiguration CreateTestConfig(string testConfigurationId)
+        {
+            TwinConfiguration twinConfiguration =
+                new TwinConfiguration
+                {
+                    Id = testConfigurationId
+                };
+
+            ChangeTrackingDictionary<string, object> deviceContent = new ChangeTrackingDictionary<string, object>();
+            deviceContent["properties.desired.deviceContent_key"] = "deviceContent_value-" + twinConfiguration.Id;
+
+            // Labels are optional but adding here due to null check failure in deserialization
+            twinConfiguration.Labels.Add("HostPlatform", Environment.OSVersion.ToString());
```

Comment:
This is one of the temporary fix for labels deserialization issue.

If this is not provided, we'll see an exception